modules/kernel/src/main/scala-2.10/notebook/ReplCalculator.scala

```diff
@@ -193,11 +193,11 @@ class ReplCalculator(
         if (queue.nonEmpty) {
           //queue could be empty if InterruptRequest was asked!
           log.debug("Dequeuing execute request current size: " + queue.size)
-          queue = queue.dequeue._2
-          queue.headOption foreach { case (ref, er) =>
```

Comment:
what this did: `remove first item from the queue, and then execute the second one, if one exists`.
sounds weird.